reusable_components_nlp/README.md

```diff
@@ -0,0 +1,989 @@
+
+# End-to-end Reusable ML Pipeline with Seldon and Kubeflow
+
+In this example we showcase how to build re-usable components to build an ML pipeline that can be trained and deployed at scale.
+
+We will automate content moderation on the Reddit comments in /r/science building a machine learning NLP model with the following components:
+
+![](img/completed-pipeline-deploy.jpg)
+
+This tutorial will break down in the following sections:
+
+1) Run all the services (Kubeflow and Seldon)
+
+2) Test and build all our reusable pipeline steps
+
+3) Use Kubeflow to Train the Pipeline and Deploy to Seldon
+
+5) Test Seldon Deployed ML REST Endpoints
+
+6) Visualise Seldon's Production ML Pipelines
+
+# Before you start
+Make sure you install the following dependencies, as they are critical for this example to work:
+
+* Helm v2.13.1+
+* A Kubernetes cluster running v1.13 or above (minkube / docker-for-windows work well if enough RAM)
+* kubectl v1.14+
+* ksonnet v0.13.1+
+* kfctl 0.5.1 - Please use this exact version as there are major changes every few months
+* Python 3.6+
+* Python DEV requirements (we'll install them below)
+
+Let's get started! 🚀🔥 We will be building the end-to-end pipeline below:
+
+![](img/kubeflow-seldon-nlp-full.jpg)
+
+
+
+```python
+!cat requirements-dev.txt
+```
+
+    python-dateutil
+    https://storage.googleapis.com/ml-pipeline/release/0.1.20/kfp.tar.gz
+    kubernetes
+    click
+    seldon_core
+    numpy
+
+
+
+```python
+!pip install -r requirements-dev.txt
```

Comment:
No, this is a `jupyter nbconvert` to README.md, the notebook is provided in the folder, I could add a note at the top pointing this out, alternatively I can only leave the intro and point a link to the notebook instead of leaving the full contents in `.md` format.

Reply:
Or do you mean specifically in regards to this command? If that is the case yeah I can just change that to a markup type and suggest for people to run it in the terminal.

Reply:
ah, I see. I'd rather have the instructions in only one place, so a link to the notebook instead of generated markdown would be better. GitHub renders notebooks cleanly so viewing isn't an issue anymore, but the downside is that version history is more obfuscated. I kind of like this approach with commands integrated, but it is different from all of our other examples. Is the expectation that a user is running a notebook server locally? Perhaps just mention that in the markdown.